target-gen/src/main.rs

```diff
@@ -86,6 +83,22 @@ enum TargetGen {
         #[clap(value_parser)]
         output: Option<PathBuf>,
     },
+    /// Generates a target yaml from a flash algorithm Rust project.
+    ///
+    /// Extracts parameters and functions from the ELF, generates the target yaml file
+    /// and runs the flash algorithm on the given attached target.
+    ///
+    /// This can be used as a cargo runner.
+    Test {
+        /// The path of the template YAML definition file.
+        /// This file plus the information of the ELF will be merged
+        /// and stored into the `definition_export_path` file.
+        template_path: PathBuf,
```

Comment:
Why is a template file needed? What information do we get from the template, that is not otherwise available?

Reply:
The template is needed for proper flash and ram range entries :)

Reply:
Basically the things that would be extracted from the `.pdsc` file and not the ELF itself :)

Reply:
But the flash size is part of the ELF, isn't it?